cli/roger_build.py

```diff
@@ -82,6 +82,12 @@ def main(self, settingObj, appObj, hooksObj, dockerUtilsObj, dockerObj, args):
             else:
                 repo = data.get('repo', args.app_name)
 
+            build_args = {}
+            if 'build-args' in data:
+                if 'environment' in data['build-args']:
```

Comment:
build args is usually used to pass in env specific args during building that would create a docker image per env. In that case, it might not make sense to use roger promote, unless someone uses the same build arg values for env1 and env2, but different for env3. In such a case, one can still use roger promote between env1 and env2.
We should not need to flag images specifically if that use build args, and also with the above use case, even if build args is used, roger promote can be used between 2 envs.